tests/integration/models/application.spec.coffee

```diff
@@ -581,3 +581,85 @@ describe 'Application Model', ->
 					.then =>
 						promise = balena.models.application.isTrackingLatestRelease(@application.id)
 						m.chai.expect(promise).to.eventually.be.true
+
+	describe 'given a multicontainer application with a single offline device', ->
+
+		givenMulticontainerApplication()
+
+		describe 'balena.models.application.getWithDeviceServiceDetails()', ->
+
+			it 'should retrieve application and it\'s devices along with service details', ->
+				balena.models.application.getWithDeviceServiceDetails(@application.id).then (applicationDetails) =>
+					m.chai.expect(applicationDetails.owns__device[0]).to.deep.match
+						device_name: @device.device_name
+						uuid: @device.uuid
+						is_on__commit: @currentRelease.commit
+						current_services:
+							web: [
+								id: @newWebInstall.id
+								service_id: @webService.id
+								image_id: @newWebImage.id
+								commit: 'new-release-commit'
+								status: 'downloading'
+								download_progress: 50
+							,
+								id: @oldWebInstall.id
+								service_id: @webService.id
+								image_id: @oldWebImage.id
+								commit: 'old-release-commit'
+								status: 'running'
+								download_progress: 100
+							]
+							db: [
+								id: @newDbInstall.id
+								service_id: @dbService.id
+								image_id: @newDbImage.id
+								commit: 'new-release-commit'
+								status: 'running'
+								download_progress: 100
+							]
+
+					# Should filter out deleted image installs
+					m.chai.expect(applicationDetails.owns__device[0].current_services.db).to.have.lengthOf(1)
+
+					# Should have an empty list of gateway downloads
+					m.chai.expect(applicationDetails.owns__device[0].current_gateway_downloads).to.have.lengthOf(0)
+
+		describe 'balena.models.application.getAllWithDeviceServiceDetails()', ->
+
+			it 'should retrieve applications and their devices, along with service details', ->
+				balena.models.application.getAllWithDeviceServiceDetails(@application.id).then (applications) =>
+					m.chai.expect(applications[0].owns__device[0]).to.deep.match
```

Comment:
Maybe you can extract the expectations of `balena.models.application.getWithDeviceServiceDetails()` into a function `itShouldBeAnApplicationWIthDeviceServiceDetails` and just call it in here, along with an extra test case about the length of the returned `applications`.